planner/select.go

```diff
@@ -296,20 +296,22 @@ func (n *selectNode) initSource() ([]aggregateNode, error) {
 	return aggregates, nil
 }
 
-func findFilteredByIndexedField(scanNode *scanNode) immutable.Option[client.FieldDescription] {
+func findFilteredByIndexedField(scanNode *scanNode) immutable.Option[client.IndexDescription] {
```

Comment:
suggestion:  Is an old name I know, but it looks like it is missing a word? I suggest renaming to `findFilteredIndexByIndexedField`, or maybe `findIndexByFilteringField`.